cmd/gitops/add/profiles/cmd.go

```diff
@@ -0,0 +1,122 @@
+package profiles
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/spf13/cobra"
+	"github.com/weaveworks/weave-gitops/cmd/internal"
+	"github.com/weaveworks/weave-gitops/pkg/flux"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	"github.com/weaveworks/weave-gitops/pkg/models"
+	"github.com/weaveworks/weave-gitops/pkg/osys"
+	"github.com/weaveworks/weave-gitops/pkg/runner"
+	"github.com/weaveworks/weave-gitops/pkg/server"
+	"github.com/weaveworks/weave-gitops/pkg/services"
+	"github.com/weaveworks/weave-gitops/pkg/services/auth"
+	"github.com/weaveworks/weave-gitops/pkg/services/profiles"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+var opts profiles.AddOptions
+
+// AddCommand provides support for adding a profile to a cluster.
+func AddCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "profile",
+		Short:         "Add a profile to a cluster",
+		SilenceUsage:  true,
+		SilenceErrors: true,
+		Example: `
+		# Add a profile to a cluster
+		gitops add profile --name=podinfo --cluster=prod --version=1.0.0 --config-repo=ssh://git@github.com/owner/config-repo.git
+		`,
+		RunE: addProfileCmdRunE(),
```

Comment:
Are we skipping the `apply` alias for add command in this PR?

Reply:
I didn't add it because it isn't really what `add profile` does (it does not reconcile), and neither `add cluster` nor `add app` have the `apply` alias. It seems like a legacy thing that we did with pctl that doesn't fit so well with the new profile implementation, but I'm open to changing this if necessary.